docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+
+As an example, of what the operational monitoring dashboards currently look like see: https://mozilla.cloud.looker.com/dashboards/operational_monitoring::bug_1715474_rollout_yandex_sponsored_tile_rollout_release_89_100?Percentile=50
+The dashboards have a tile for each metric which shows the values of all available branches for the percentile currently selected as filter.
+
+## Constraints
+
+There are a couple of constraints when checking for critical changes in OpMon:
+* **Cost:** We will want to keep the cost low since there will be a lot of dashboards (for example, we create one for each rollout)
+* **Computation Speed:** The operational monitoring dashboards get updated on a daily basis and there are usually dozens of dashboards that need to get updated. Ideally, the checks for each dashboard and day finish within minutes rather than hours.
+* **Minimize False Positives:** Sending alerts when nothing is wrong too many times results in users simply ignoring alerts at some point
+
+## Monitor Types
+
+When talking to users there are a few different cases when they would like to receive an alert:
+* **Large differences between branches:** Whenever the confidence intervals of different branches for a specific metric no longer overlap, it indicates that there is potentially some significant difference.
+* **Thresholds:** Comparing the values of a metric to a user-defined threshold.
+* **Deviation from historical data:** Detect anomalous behaviour of a metric based on previously collected data.
+
+### Large differences between branches
+
+The OpMon dashboards show the values for specific metrics as a line chart with confidence intervals. Each line represents the metric values for a different branch. Whenever the confidence intervals of the branches do not overlap, it is considered a critical change. See:
+
+<img src="./images/alerting_branch_differences.png" width="500"/>
+
+These checks need to be manually configured for specific metrics/probes. The OpMon configuration files will be extended to something like this:
+
+```toml
+[alert]
+
+[alert.crashes] # new alert
+type = "ci"
+probes = [      # list of metrics to check
+    "startup_crashes", 
+    "main_crashes", 
+    "oom_crashes"
+]
+percentiles = [50, 90]  # percentiles that should be considered
+```
+
+
+### Thresholds
+
+In some cases the expected value of a metric is known and any large deviation from that expected value is considered a critical change. Fixed thresholds can be used to specify when a value is too large or too low. See:
+
+<img src="./images/alerting_thresholds.png" width="500"/>
+
+These checks need to be manually configured for specific metrics/probes. The OpMon configuration files will be extended to something like this:
+
+```toml
+[alert]
+
+[alert.crash_diffs] # new alert
+type = "threshold"
+precentiles = [50, 90] # percentiles that should be considered
```

Comment:
I'm concerned about the current method for comparing percentiles. Currently we're using the [empirical CDF method with jackknife resampling](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/udf_js/jackknife_percentile_ci/udf.sql) which, though it has advantages, very often results in zero-width confidence intervals.  A basic summary of the issue is that we'll only really get an estimate of variability if we have only one or two samples with that percentile value. For example, if the 90th percentile is 7 and we have 5 samples with 7, our CI will end up looking like (7,7) so then any change from 7 will be "statistically significant". 

This is why, in many of the example dashboards we've built so far, we mostly have lower bounds equal to upper bounds with the occasional massive spike.